Assets/Scripts/Game/Guilds/GuildManager.cs

```diff
@@ -178,7 +178,11 @@ private IGuild CreateGuildObj(FactionFile.GuildGroups guildGroup, int variant =
                         return (IGuild)Activator.CreateInstance(guildType, new object[] { KnightlyOrder.GetOrder(variant) });
 
                     default:
-                        return (IGuild)Activator.CreateInstance(guildType);
+                        try {
+                            return (IGuild)Activator.CreateInstance(guildType, new object[] { variant });
```

Comment:
I'd rather see something with `if(guildType.GetConstructor(new Type[] { typeof(int) }) != null)` than exceptions. There is the performance concern, sure, but also I set my debugger to always break when an exception is thrown in user code, so I'd have to mute that exception type entirely
Suggestion comes from: https://stackoverflow.com/a/19611584/2356414

Reply:
Originally I was going to require all custom guild classes implement default and variant param constructors but then I figured this could be handled by a try catch so mod authors were not required to do that even though I'd still recommend it. I can change it to check rather than handle the exception like you suggest if you like.

Reply:
Yes, of course it should be optional, for backward compatibility and general convenience. It's really about not relying on exceptions if we can avoid it